lava-v2-jobs-from-api.py

```diff
@@ -58,17 +70,33 @@ def main(args):
     }
 
     print "Working on kernel %s/%s" % (tree, branch)
-    url = urlparse.urljoin(api, ("/build?job=%s&kernel=%s&git_branch=%s&status=PASS&arch=%s" % (tree, git_describe, branch, arch)))
+    url_params = urllib.urlencode({
+        'job': tree,
+        'kernel': git_describe,
+        'git_branch': branch,
+        'status': 'PASS',
+        'arch': arch,
+    })
+    url = urlparse.urljoin(api, 'build?{}'.format(url_params))
     print "Calling KernelCI API: %s" % url
-    response = requests.get(url, headers=headers)
-    if response.status_code != 200:
-        print "Error calling KernelCI API"
-        print response
-        print response.content
-        sys.exit(1)
-    data = json.loads(response.content)
-    builds = data['result']
+
+    builds = None
+    poll_seconds = 10
+    loops = 600 / poll_seconds # time out after 10 minutes
+    loop = 0
+    while not builds and loop < loops:
```

Comment:
This was discussed on IRC, it's a real issue.  When a boot report has been sent to the KernelCI backend, an asynchronous task is being run (with Celery).  So the boot data is not in the database at the end of the request, it only gets added later on.  It is usually fast but there's nothing to guarantee when the boot data will actually be in the database.

I could hit this issue regularly on my set-up running in a VM, it could sometimes take a few seconds between the end of the "boot POST" request and when the data is actually available in the database.  Calling the lava-v2-jobs-from-api.py script too early can cause some kernel builds to not be boot tested.